master/internal/db/postgres_tasks.go

```diff
@@ -120,6 +120,17 @@ func (db *PgDB) DeleteAllocationSession(allocationID model.AllocationID) error {
 	return err
 }
 
+// UpdateAllocationState stores the latest task state and readiness.
+func (db *PgDB) UpdateAllocationState(allocationID model.AllocationID,
```

Comment:
after talking yesterday, I think these fields should be added to the `model.Allocation` struct, updated in the copy of it nested in the allocation and it should be received here.

Reply:
so would this replace `UpdateAllocationState` with `UpdateAllocation`?